sources/packages/web/src/views/student/financial-aid-application/FullTimeApplication.vue

```diff
@@ -43,63 +87,129 @@ export default {
       } catch (error) {
-        console.error(error);
+        toast.error("Unexpected error!", "An unexpected error happen.");
+      } finally {
+        savingDraft.value = false;
       }
     };
 
-    onMounted(async () => {
-      if (props.id) {
-        const dataToLoad = await ApiClient.Application.getApplicationData(
-          props.id,
+    // Execute the final submission of the student application.
+    const submitApplication = async (args: any, form: any) => {
+      submittingApplication.value = true;
+      try {
+        const associatedFiles = formioUtils.getAssociatedFiles(form);
+        await ApplicationService.shared.submitApplication(props.id, {
+          programYearId: props.programYearId,
+          data: args,
+          associatedFiles,
+        });
+        toast.success(
+          "Application saved!",
+          "Application submitted with success.",
         );
-        initialData.value = dataToLoad;
-      } else {
-        //Get the student info from api call
-        const studentInfo = await StudentService.shared.getStudentInfo();
-
-        // TODO: Move formatted address to a common place in Vue app or API.
-        // Adjust the spaces when optional fields are not present.
-        const address = studentInfo.contact;
-        const formattedAddress = `${address.addressLine1} ${address.addressLine2} ${address.city} ${address.provinceState} ${address.postalCode}  ${address.country}`;
-
-        initialData.value = {
-          studentGivenNames: studentInfo.firstName,
-          studentLastName: studentInfo.lastName,
-          studentGender: studentInfo.gender,
-          studentDateOfBirth: studentInfo.birthDateFormatted,
-          studentPhoneNumber: studentInfo.contact.phone,
-          studentHomeAddress: formattedAddress,
-          studentEmail: studentInfo.email,
-        };
+      } catch (error) {
+        toast.error("Unexpected error!", "An unexpected error happen.");
+      } finally {
+        submittingApplication.value = false;
       }
-    });
+    };
 
     // Components names on Form.IO definition that will be manipulated.
     const LOCATIONS_DROPDOWN_KEY = "selectedLocation";
     const PROGRAMS_DROPDOWN_KEY = "selectedProgram";
     const OFFERINGS_DROPDOWN_KEY = "selectedOffering";
     const SELECTED_OFFERING_DATE_KEY = "selectedOfferingDate";
-
     const formLoaded = async (form: any) => {
+      applicationWizard = form;
+      // Disable internal submit button.
+      formioUtils.disableWizardButtons(applicationWizard);
+      applicationWizard.options.buttonSettings.showSubmit = false;
+      // Handle the navigation using the breadcrumbs.
+      applicationWizard.on("wizardPageSelected", (page: any, index: number) => {
+        isFirstPage.value = index === 0;
+        isLastPage.value = applicationWizard.isLastPage();
+      });
+      // Handle the navigation using next/prev buttons.
+      const prevNextNavigation = (navigation: any) => {
```

Comment:
can we add an interface for `navigation`. or the type is uncertain?

Reply:
We can, but this is a Form.IO event (which is not a proper justification 😄). In general, we can add interfaces and types for all Form.IO items that we have been interacting with, but we are not doing it so far.
I would recommend that we create a ticket to go back and create the types for all others areas also. What do you think?

Reply:
Talked with @ann-aot and the idea is to start the Form.IO types mapping with this one and keeping doing it as we go.

Reply:
yes , I agree @andrewsignori-aot

Reply:
Added the WizardNavigationEvent interface to the newly created formio.ts types.
@guru-aot and @dheepak-aot, whatever we have an opportunity to map the Form.IO types that are defined as "any" let's please do it. Please let us know if you have any questions about it.